src/graph/graphcontext.c

```diff
@@ -11,11 +11,25 @@
 #include "../query_ctx.h"
 #include "../redismodule.h"
 #include "../util/rmalloc.h"
+#include "../util/thpool/thpool.h"
 #include "serializers/graphcontext_type.h"
 
 // Global array tracking all extant GraphContexts (defined in module.c)
+extern threadpool _thpool;
 extern GraphContext **graphs_in_keyspace;
 
+// Forward declarations.
+static void _GraphContext_Free(void *arg);
+
+static inline void _GraphContext_IncreaseRefCoun(GraphContext *gc) {
+	__atomic_fetch_add(&gc->ref_count, 1, __ATOMIC_RELAXED);
+}
+
+static inline void _GraphContext_DecreaseRefCoun(GraphContext *gc) {
+	if(__atomic_sub_fetch(&gc->ref_count, 1, __ATOMIC_RELAXED) < 0)
```

Comment:
This works fine for me on GCC and Clang, so I think we're good! An explanatory comment here would be a good idea, though.